backend/session 4/app.py

```diff
@@ -0,0 +1,334 @@
+from flask import Flask, request, session, jsonify
+from models import db, User, Task, Note, Abroad_blogs
+from flask_cors import CORS
+from flask_bcrypt import Bcrypt
+from datetime import timedelta
+from flask_jwt_extended import (
+    JWTManager,
+    create_access_token,
+    jwt_required,
+    get_jwt_identity,
+)
+import jwt
+
+app = Flask(__name__)
+CORS(app)
+jwt = JWTManager(app)
+app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///databse.db"
+app.config["SECRET_KEY"] = "secret_key"
+db.init_app(app)
+bcrypt = Bcrypt(app)
+
+with app.app_context():
+    db.create_all()
+
+
+@app.route("/signup", methods=["POST"])
+def registration():
+    json = request.get_json()
+    hashed_password = bcrypt.generate_password_hash(json["password"]).decode("utf-8")
+    user = User(
+        email=json["email"],
+        password_hash=hashed_password,
+        username=json["username"],
+        age=json["age"],
+        school=json["school"],
+        gender=json["gender"],
+    )
+    user_db_email = User.query.filter_by(email=user.email).first()
+    user_db_email_username = User.query.filter_by(username=user.username).first()
+    if not (user_db_email is None):
+        return jsonify("Email already saved"), 409
+    if not (user_db_email_username is None):
+        return jsonify("Username is already saved "), 409
+    db.session.add(user)
+    db.session.commit()
+    return jsonify("Signed up successfully"), 200
+
+
+@app.route("/login", methods=["POST"])
+def login():
+    json = request.get_json()
+    email = json["email"]
+    password = json["password"]
+    user = User.query.filter_by(email=email).first()
+    if user and bcrypt.check_password_hash(user.password_hash, password):
+        access_token = create_access_token(
+            identity=email,
+            expires_delta=timedelta(weeks=52),
+        )
+        return jsonify(access_token=access_token), 200
+    return jsonify({"msg": "Invalid email or password"}), 401
```

Comment:
try handling the case of 
if user is not registered on system already.
You have 2 cases of error: either user wrote wrong email or password, or user doesn't exist.